kuebiko/pipeline/nspl/flow.py

```diff
@@ -0,0 +1,215 @@
+    ## a long time to build/install or it has many transient dependencies
+    ## that may cause conflicts with our main environment.
+    ## Because it's unlikely to be needed to do the pure analysis (which
+    ## occurs in our project's conda environment), we can use
+    ## [`@conda_base`](https://docs.metaflow.org/metaflow/dependencies) to
+    ## create a separate Conda environment and run our flow inside that.
+    ## This minimises the time to install and chance of dependency conflicts
+    ## in the environment that everyone must install, not just the
+    ## machine/person running this pipeline.
+    ## In reality, it's best to **stick to using your project's main environment
+    ## and only use Metaflow's conda decorators when you definitely need them**.
+    libraries={
+        "pandas": "1.1.0",
+        "selenium": "3.141.0",
+        "requests-cache": "0.4.13",
+    }
+)
+@project(name="kuebiko")
+## A flows docstring is one place to document (or refer to)
+## shortcomings in the data.
+class NsplLookup(FlowSpec):
+    """Lookups from postcode to Local Authority Districts and lat-long.
+
+    Uses National Statistics Postcode Lookup (NSPL).
+    Excludes postcodes that do not have an assigned output area (OA),
+    and therefore no Local Authority District coding.
+    Postcodes in the Channel Islands and Isle of Man both have lat-longs of
+    `(99.999999, 0.000000)` and have pseudo Local Authorit district codes of
+    "L99999999" and "M99999999" respectively, this does not match up with
+    other datasets and therefore have been excluded.
+
+    Attributes:
+        geoportal_url: Full URL of dataset in gov.uk geoportal
+        download_url: Download URL of dataset
+        laua_names: Lookup from laua code to name
+        laua_year: Year LAUA names and codes correspond to
+        pcd_laua: Lookup from postcode to LAUA
+        pcd_latlong: Lookup from postcode to latitude and longitude
+    """
+
+    ## We can use parameters to allow changes to the flows behaviour without,
+    ## requiring changes to the code. In this case, we can parameterise the
+    ## particular version of the dataset we're using, falling back on a
+    ## sensible default.
+    geoportal_dataset = Parameter(
+        "geoportal-dataset",
+        help=f"Name of dataset in URL at {GEOPORTAL_URL_PREFIX}",
+        required=True,
+        type=str,
+        default="national-statistics-postcode-lookup-august-2021",
+    )
+
+    ## **If your flow takes more than a minute or two to run then you should
+    ## always add a parameter `test-mode`** which will run your flow in test
+    ## mode, e.g. fetch/run a subset of the data.<br>
+    ## This facilitates both efficient code review and automated testing.
+    ## We recommend that by default `test-mode` is either:
+    ##
+    ## - The logical not of `current.is_production`
+    ##   (so that this is evaluated at the right time, it must be a callable)
```

Comment:
Sorry, I'm not sure I understand "so that this is evaluated at the right time, it must be a callable"